packages/auth/src/accessToken/service.ts

```diff
@@ -81,6 +89,21 @@ async function introspect(
     if (!token.grant || isRevokedGrant(token.grant)) {
       return undefined
     }
+    if (access) {
+      for (const accessItem of access) {
+        const { access: grantAccess } = token.grant
+        if (
+          !grantAccess.find((grantAccessItem) =>
+            compareRequestAndGrantAccessItems(
+              accessItem,
+              toOpenPaymentsAccess(grantAccessItem)
+            )
+          )
+        ) {
+          return undefined
+        }
```

Comment:
to understand this correctly, in this implementation, the `access` will consist of an array with a single item of { `identifier` `access` `type` }. We then compare this object to the existing grantAccessItem. 
And even though `accessItem` will never have `limits`, `compareRequestAndGrantAccessItems` checks if all the other keys are equivalent, correct?

Reply:
Yeah that's pretty much it.